osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs

```diff
@@ -111,6 +111,30 @@ private void load(BeatmapManager manager, BeatmapSetOverlay beatmapOverlay)
             };
         }
 
+        private readonly List<DrawableCarouselBeatmap> beatmapsToCheck = new List<DrawableCarouselBeatmap>();
+
+        public void AssureBeatmapPositionValid(DrawableCarouselBeatmap beatmap) => beatmapsToCheck.Add(beatmap);
+
+        public void ClearAssurance() => beatmapsToCheck.Clear();
```

Comment:
If it the beatmap set panels weren't entirely decoupled from the difficulty panels before the introduction of this list, this could have been a `osu.Framework.Caching.Cached` (*not* the DI attribute), but as things stand this list is needed in this current implementation.

Reply:
I was confused about this when I first read it. The panels actually need to be decoupled and to my knowledge always have been. The logic to remove offscreen panels from the scene graph relies on all the panels being decoupled. Otherwise removing one panel might remove stuff some others rely on. 
So yeah, this can't use caching.